### DISCUSSION TITLE
Lack of AMQP documentation

### DISCUSSION THREAD

**Discussion by fabiante** [go-kit/kit#1146] [Q&A]
Hey there,

I am sorry to admit, that I don't get at all, on how to use the AMQP transport for subscribing to a queue.

I have now been experimenting quiet some time now, but I just don't know how to simply listen to a queue and use the amqp transport.

Could someone please provide me with a full example?

I would be happy to derive an example from that and publish it via a PR.

**Comment by chaudharisuresh997** [comment]
At least give contract wherer we can create these for Apache kafka and activemq

**Comment by chaudharisuresh997** [comment]
@peterbourgon

**Comment by teploff** [comment]
Relevant question. @peterbourgon, can you help us?

**Comment by gadieichhorn** [comment]
did anyone found an example? 
would be great if a  link could be shared :)

**Comment by teploff** [comment]
Unfortunately no.
I just wrote custom client and injected it.

**Comment by kallydev** [comment]
> did anyone found an example?
> would be great if a link could be shared :)

Ref https://github.com/go-kit/kit/pull/746#issuecomment-439225894

**Reply by kallydev** [reply]
https://github.com/cloudpbx/kit/tree/amqp/examples/stringsvc4/amqp

**Comment by VinhVu95** [comment]
Can someone give an example in RabbitMQ please?
